src/cljs/rems/status_modal.cljs

```diff
@@ -1,67 +1,118 @@
-   (case status
-     nil ""
-     :pending [spinner/big]
-     :saved [:div [:i {:class ["fa fa-check-circle text-success"]}] (text :t.form/success)]
-     :failed [:div [:i {:class "fa fa-times-circle text-danger"}]
-              (str (text :t.form/failed)
-                   (when (:key error)
-                     (str ": " (text (:key error))))
-                   (when-let [text (:status-text error)]
-                     (str ": " text))
-                   (when-let [text (:status error)]
-                     (str " (" text ")")))])])
+(defn- format-errors [errors]
+  [:div (for [error errors]
+          [:p
+           (when (:key error)
+             (str (text (:key error))))
+           (when (:type error)
+             (str (text (:type error))))
+           (when-let [text (:status-text error)]
+             (str text))
+           (when-let [text (:status error)]
+             (str " (" text ")"))])])
+
+(defn- status-widget [success? error-content]
+  (cond
+    (and (not success?) (not error-content)) [spinner/big]
+    success? [:p#status-success [:i {:class ["fa fa-check-circle text-success"]}] (text :t.form/success)]
+    error-content [:div [:p#status-failed [:i {:class "fa fa-times-circle text-danger"}] (text :t.form/failed)]
+                   error-content]))
+
+(rf/reg-event-db ::set-state (fn [db [_ state]] (assoc db ::state state)))
+(rf/reg-event-db ::merge-state (fn [db [_ state]] (update db ::state deep-merge state)))
+(rf/reg-sub ::state (fn [db _] (::state db)))
 
 (defn status-modal
   "Modal component showing the status of an action.
 
-  `:status` - Show spinner while `:pending`. Either `:saved` or `:failed`
-            status is shown with the internal status-widget that
-            may show the `:error` contents.
-  `:description` - the title of the modal.
-  `:content` - additional content to show after the status widget.
-  `:error` - error that may contain :key, :status and :status-text
-           like translated errors or http errors
-  `:on-close` - callback is called when the modal wants to close itself"
-  [{:keys [description status error content on-close]}]
-  [modal/notification {:title description
-                       :content [:div [status-widget status error] content]
-                       :on-close on-close
-                       :shade? true}])
+  `:result`        - Either {:success? true} or {:error ...} {:errors ...}
+                     Show spinner while neither.
+  `:title`         - title of the modal, i.e. name of the operation
+  `:content`       - additional content to show after the status widget
+  `:error`         - error that may contain `:key`, `:type`, `:status` and `:status-text`
+                     like translated errors or http errors
+  `:error-content` - content to show instead of generated content from errors
+  `:on-close`      - callback is called when the modal wants to close itself"
+  [initial-state]
```

Comment:
hmm so `initial-state` is only used for the guide stuff. should there be an explicit 0-arity version?

Reply:
Yes I'll make it explicitly optional. ClojureScript doesn't care but it's better documented that way.

Reply:
The documentation is useful for the other fns.